source/indexes.txt

```diff
@@ -24,12 +24,146 @@ Optimize Queries by Using Indexes
    
    /indexes/single-field-index
    /indexes/compound-index
+   /indexes/atlas-search-index
 
 Overview
 --------
 
 On this page, you can see copyable code examples that show how to manage
 different types of indexes by using the {+driver-long+}.
 
-.. TODO
+To use an example from this page, copy the code example into the sample 
+application or your own application. Be sure to replace all placeholders in the 
+code examples, such as ``<connection string URI>``, with the relevant values for 
+your MongoDB deployment.
 
+Sample Application
+~~~~~~~~~~~~~~~~~~
+
+You can use the following sample application to test the code on this page. To
+use the sample application, perform the following steps:
+
+1. Ensure you have the {+driver-short+} installed in your project. See the 
+   :ref:`scala-quick-start-download-and-install` guide to learn more.
+
+#. Copy the following code and paste it into a new ``.scala`` file.
+
+#. Copy a code example from this page and paste it on the specified lines in the
+   file.
+
+.. literalinclude:: /includes/indexes/starter-indexes.scala
+   :language: scala
+   :emphasize-lines: 18-21
+
+Single Field Index
+------------------
+
+The following example creates an ascending index on the specified field:
+
+.. literalinclude:: /includes/indexes/single-field.scala
```

Comment:
S: for the landing pages, we usually use placeholder values and keep the examples very generic ([PHP](https://www.mongodb.com/docs/php-library/current/indexes/), [C++](https://www.mongodb.com/docs/languages/cpp/cpp-driver/current/indexes/), etc). I'd suggest creating a separate usage examples file instead of using the specific examples from the guides - here's the [PHP file](https://github.com/mongodb/docs-php-library/blob/master/source/includes/usage-examples/index-code-examples.php)